### PR TITLE
Performance update for binomial test

### PR DESCRIPTION
**Pull request by zhut19**
## What does the code in this PR do / what does it improve?
Speeds up the non-discrete binomial test p-value computation for S1 area fraction top

## Can you briefly describe how it works?
Use secant method instead of grid search for finding the mode and j-value from binomial pmf

### CODE REVIEW

**Comment by JoranAngevaare** [comment]
Thanks @zhut19! Do you have a feeling for the performance upgrade? Do we meed to increment the plugin version?

I think @GiovanniVolta can best comment on testing that the results are the same.

**Comment by GiovanniVolta** [comment]
Hello everyone, me and Tyaniu already discuss it. This improvement looks very promising. However, I would like to test it and compare the two method. I will do it as soon as possible.

**Comment by WenzDaniel** [comment]
As mentioned by Joran some benchmark would be interesting. Can be posted in the Team B channel in slack.

File: straxen/plugins/event_patternfit.py [hunk 1]
```diff
@@ -339,162 +338,122 @@ def neg2llh_modpoisson(mu=None, areas=None, mean_pe_photon=1.0):
...
     return pval
 
-def _s1_area_fraction_top_probability(aft_prob, area_tot, area_fraction_top, mode='continuous'):
-    '''
-    Wrapper that does the S1 AFT probability calculation for you
-    '''
-    
+
+@np.vectorize
+@numba.njit
```
Comment (WenzDaniel) [inline straxen/plugins/event_patternfit.py:452]: Is not it possible to use numbas vectorize routine? See removed line 344.

**Review by WenzDaniel — COMMENTED** [review]
From a technical point of view it looks fine to me. Although I have not checked if the change makes any logical sense.

**Comment by GiovanniVolta** [comment]
Hey @WenzDaniel, besides the issue I pointed out in my review, I think @zhut19 proposal is better than what we are doing right now

**Comment by WenzDaniel** [comment]
So we can go ahead and merge?

**Comment by GiovanniVolta** [comment]
>So we can go ahead and merge?

No, my review has not been addressed yet

**Comment by WenzDaniel** [comment]
Convert it to a draft until Giovannis comments are addressed.

**Review by GiovanniVolta — CHANGES_REQUESTED** [review]

**Comment by GiovanniVolta** [comment]
Sorry guys, I am terrible with GitHub. I didn't submit my review

**Review by GiovanniVolta — APPROVED** [review]
Even if there is room for improvement, I am happy with this. I am making a summary note that describe how we are computing the p-value. I hope it will be useful for the later.
Could you please Tianyu update the version number?

**Comment by WenzDaniel** [comment]
@zhut19  as soon as you bump the version I will make the new release.

**Pull request merged**